src/streamlink/plugins/twitch.py

```diff
@@ -137,18 +137,45 @@
 
 Segment = namedtuple("Segment", "uri duration title key discontinuity scte35 byterange date map")
 
+LOW_LATENCY_MAX_LIVE_EDGE = 2
+
+
+def parse_condition(attr):
+    def wrapper(func):
+        def method(self, *args, **kwargs):
+            if hasattr(self.stream, attr) and getattr(self.stream, attr, False):
+                func(self, *args, **kwargs)
+        return method
+    return wrapper
+
 
 class TwitchM3U8Parser(M3U8Parser):
+    def __init__(self, base_uri=None, stream=None, **kwargs):
+        M3U8Parser.__init__(self, base_uri, **kwargs)
+        self.stream = stream
+
+    @parse_condition("disable_ads")
     def parse_tag_ext_x_scte35_out(self, value):
         self.state["scte35"] = True
 
     # unsure if this gets used by Twitch
+    @parse_condition("disable_ads")
     def parse_tag_ext_x_scte35_out_cont(self, value):
         self.state["scte35"] = True
 
+    @parse_condition("disable_ads")
     def parse_tag_ext_x_scte35_in(self, value):
         self.state["scte35"] = False
 
+    @parse_condition("low_latency")
+    def parse_tag_ext_x_twitch_prefetch(self, value):
+        segments = self.m3u8.segments
+        if segments:
+            last_segment = segments[-1:][0]
+            new_segment = last_segment._replace(uri=self.uri(value))
+            segments.append(new_segment)
+            log.debug("Prefetching segment {0}".format(self.m3u8.media_sequence + len(segments) - 1))
```

Comment:
to much spam and duplicate messages

it also doesn't prefetch anything here, it just queue it to a list so this is a bit misleading

```
[stream.hls][debug] Reloading playlist
[plugin.twitch][debug] Prefetching segment 3405
[plugin.twitch][debug] Prefetching segment 3406
[stream.hls][debug] Adding segment 3406 to queue
[stream.hls][debug] Download of segment 3405 complete
[stream.hls][debug] Reloading playlist
[plugin.twitch][debug] Prefetching segment 3406
[plugin.twitch][debug] Prefetching segment 3407
[stream.hls][debug] Adding segment 3407 to queue
[stream.hls][debug] Download of segment 3406 complete
[stream.hls][debug] Reloading playlist
[plugin.twitch][debug] Prefetching segment 3407
[plugin.twitch][debug] Prefetching segment 3408
[stream.hls][debug] Adding segment 3408 to queue
[stream.hls][debug] Download of segment 3407 complete
[stream.hls][debug] Reloading playlist
[plugin.twitch][debug] Prefetching segment 3408
[plugin.twitch][debug] Prefetching segment 3409
[stream.hls][debug] Adding segment 3409 to queue
[stream.hls][debug] Download of segment 3408 complete
[stream.hls][debug] Reloading playlist
[plugin.twitch][debug] Prefetching segment 3409
[plugin.twitch][debug] Prefetching segment 3410
[stream.hls][debug] Adding segment 3410 to queue
[stream.hls][debug] Download of segment 3409 complete
[stream.hls][debug] Reloading playlist
[plugin.twitch][debug] Prefetching segment 3410
[plugin.twitch][debug] Prefetching segment 3411
```

Reply:
True, I'll remove it again